controller/metrics/metrics.go

```diff
@@ -160,12 +162,12 @@ func NewMetricsServer(addr string, appLister applister.ApplicationLister, appFil
 
 	mux := http.NewServeMux()
 	registry := NewAppRegistry(appLister, appFilter, appLabels)
-	registry.MustRegister(depth, adds, latency, workDuration, unfinished, longestRunningProcessor, retries)
+
 	mux.Handle(MetricsPath, promhttp.HandlerFor(prometheus.Gatherers{
 		// contains app controller specific metrics
 		registry,
-		// contains process, golang and controller workqueues metrics
-		prometheus.DefaultGatherer,
+		// contains workqueue metrics, process and golang metrics
+		ctrl_metrics.Registry,
```

Comment:
This adds the rest_client metrics like the following, so far no idea how to unregister them. I think the cardinality of host is high and I feel they might not be necessary (they are also currently present on the applciationset controller when using the default metrics handler for k8s controller)

```
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="0.005"} 0
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="0.025"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="0.1"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="0.25"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="0.5"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="1"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="2"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="4"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="8"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="15"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="30"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="60"} 2
rest_client_request_duration_seconds_bucket{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET",le="+Inf"} 2
rest_client_request_duration_seconds_sum{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET"} 0.023560624000000002
rest_client_request_duration_seconds_count{host="https://dev.argocd.svc:49999/apis/networking.k8s.io/v1?timeout=32s",verb="GET"} 2
```

Reply:
Yep. It introduces a whole lot of new metrics including cert, rest client and load balancer metrics. Wondering if we should go ahead with this change..?

Reply:
Fixed by https://github.com/kubernetes-sigs/controller-runtime/pull/2299. Bumping to https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.14.7